core/src/main/java/org/apache/iceberg/OverwriteData.java

```diff
@@ -27,10 +30,15 @@
 import org.apache.iceberg.expressions.StrictMetricsEvaluator;
 
 public class OverwriteData extends MergingSnapshotProducer<OverwriteFiles> implements OverwriteFiles {
-  private boolean validateAddedFiles = false;
+  private boolean overwriteByRowFilter;
+  private boolean validateAddedFilesMatchRowFilter;
+  private boolean deleteFilesByPath;
+  private Long readSnapshotId;
+  private Expression conflictingAppendsRowFilter;
 
   protected OverwriteData(TableOperations ops) {
     super(ops);
+    failMissingDeletePaths();
```

Comment:
Is this always required? It seems like it might only be needed when validating conflicting operations.